httpcache.go

```diff
@@ -34,7 +34,7 @@ type Cache interface {
 	// set to true if the value isn't empty
 	Get(key string) (responseBytes []byte, ok bool)
 	// Set stores the []byte representation of a response against a key
-	Set(key string, responseBytes []byte)
+	Set(key string, responseBytes []byte, duration time.Duration)
```

Comment:
```
// Set stores the []byte representation of a response against a key
```

Could you please better document the new parameter? This is the `Cache` interface definition, and it's very important both for users to know what `duration` they should set, and for implementations of the interface to know what's expected of them.